### PR TITLE
fix TUN networking in LXC containers

### PR DESCRIPTION
**Pull request by volth**
in LXC container /dev/net/tun is pre-available, "dev-net-tun.device" always fails

From the comment https://github.com/NixOS/nixpkgs/issues/9735#issuecomment-500164017

### CODE REVIEW

File: nixos/modules/system/boot/systemd.nix [hunk 1]
```diff
@@ -73,7 +73,7 @@ let
       "systemd-journald.service"
       "systemd-journal-flush.service"
       "systemd-journal-catalog-update.service"
-      "systemd-journald-audit.socket"
+      ] ++ (optional (!config.boot.isContainer) "systemd-journald-audit.socket") ++ [
```
Comment (flokli) [inline nixos/modules/system/boot/systemd.nix:76]: This seems to be really not related to this PR. Can you explain what this is about?

File: nixos/modules/system/boot/systemd.nix [hunk 2]
```diff
@@ -101,7 +101,7 @@ let
       "dev-hugepages.mount"
       "dev-mqueue.mount"
       "sys-fs-fuse-connections.mount"
-      "sys-kernel-config.mount"
+      ] ++ (optional (!config.boot.isContainer) "sys-kernel-config.mount") ++ [
```
Comment (flokli) [inline nixos/modules/system/boot/systemd.nix:104]: It might be /this/ gets pre-mounted in some lxc installations, but not all containers?

Comment (volth) [reply nixos/modules/system/boot/systemd.nix:104]: How to fix it?
Add `config.boot.isLXCContainer`?
Or configure/patch `systemd` for not stopping on `sys-kernel-config.mount` failure?

File: nixos/modules/system/boot/systemd.nix (hunk 1 above)
Comment (volth) [reply nixos/modules/system/boot/systemd.nix:76]: It is part of those LXC fixes from https://github.com/NixOS/nixpkgs/issues/9735#issuecomment-500164017
PR's title does not include for brevity.

Comment (flokli) [reply nixos/modules/system/boot/systemd.nix:76]: It wouldn't have hurt to explain it in the commit message IMHO.

File: nixos/modules/system/boot/systemd.nix (hunk 2 above)
Comment (flokli) [reply nixos/modules/system/boot/systemd.nix:104]: I think it'd be best to have systemd work in both cases. We don't always know the container environment that's used to spin the image up - having systemd work in both environments might be a better idea.

Can you check if this is a problem in other distros as well? In that case, I'd propose sending a PR to upstream systemd.
